bestpractices/content/passwords.md

```diff
@@ -0,0 +1,64 @@
+
+1. The value of using a password manager for both personal and project credentials.  
+2. How to scan for accidentally committed secrets using open source tools like [trufflehog](https://github.com/trufflesecurity/trufflehog).  
+3. Techniques for setting up pre-commit hooks and CI scans to ensure secrets do not leak into your repositories.
+
+---
+
+## Risks of Insecure Password Management
+
+Hardcoded passwords, API keys, and other secrets in your repository can expose your project to risks like:
+
+- **Account Takeover**: Exposed credentials can allow unauthorized access to critical services (e.g., package registries, CI/CD systems, or infrastructure).  
+- **Reputation Damage**: A security breach can harm your project's reputation, deterring potential collaborators or users.
+- **User Supply Chain Risk**: With unauthorized access, attackers can push harmful content to your users. Most often, this is simply defacement or malicious binaries (like cryptojackers), but can sometimes be more insidious.
+
+Ensuring that passwords and other sensitive credentials remain secure is essential for any project that wants to maintain trust and protect both its developers and users.
+
+---
+
+## Best Practices for Secret Handling
+
+1. **Never store passwords or tokens directly in your repository.** Use something like GitHub Secrets to manage your secrets during CI/CD.
+2. **Limit Credential Scope**: Give each service or token the least privileges possible. Avoid reusing passwords across multiple services.  
+3. **Rotate Credentials Regularly**: If you accidentally expose a secret, be prepared to revoke or rotate it immediately.  
+4. **Educate Contributors**: Let your community know about these security measures. Provide clear guidelines for how they should handle secrets.
+
+---
+
+## Using a Password Manager
+
+One of the easiest ways to ensure secrets stay secure is to use a password manager. Password managers offer:
+
+- **Encrypted Storage**: Passwords are stored in an encrypted vault, reducing the risk of accidental exposure.  
+- **Secure Sharing**: Share credentials safely among project maintainers without resorting to email or plaintext messages.  
+- **Strong, Unique Passwords**: Automatically generate strong passwords for each service to minimize the damage if one account is compromised.
+
+If you’re looking for a managed solution, [1Password](https://github.com/1Password/for-open-source) provides a **free Teams account for Open Source Projects**. An open source alternative is [Bitwarden](https://github.com/bitwarden/).
+
+---
+
+## Scanning for Secrets
+
+Even with a password manager, mistakes can happen. You or a contributor might accidentally commit a password, API key, or other credentials to the repository. Fortunately, tools like [trufflehog](https://github.com/trufflesecurity/trufflehog) can help detect these secrets before they make their way into production or public repositories.
+
```

Comment:
I think we should also mention that GitHub also has its secret scanning service, which is available for free for public repositories: https://docs.github.com/en/code-security/secret-scanning/enabling-secret-scanning-features (and paid for private repositories).

The last time I checked, it used to be enabled by default – I'm a bit surprised that's no longer the case (or I'm misremembering).